sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/http/policy/BearerTokenAuthenticationChallengePolicy.java

```diff
@@ -32,70 +21,41 @@
 public class BearerTokenAuthenticationChallengePolicy implements HttpPipelinePolicy {
     private static final String AUTHORIZATION_HEADER = "Authorization";
     private static final String BEARER = "Bearer";
-    public static final Pattern AUTHENTICATION_CHALLENGE_PATTERN =
-        Pattern.compile("(\\w+) ((?:\\w+=\".*?\"(?:, )?)+)(?:, )?");
-    public static final Pattern AUTHENTICATION_CHALLENGE_PARAMS_PATTERN =
-        Pattern.compile("(?:(\\w+)=\"([^\"\"]*)\")+");
     public static final String WWW_AUTHENTICATE = "WWW-Authenticate";
-    public static final String CLAIMS_PARAMETER = "claims";
 
-    private final TokenCredential credential;
-    private final String[] scopes;
-    private final Supplier<Mono<AccessToken>> defaultTokenSupplier;
-    private final AccessTokenCache cache;
+    private final AccessTokenCacheImpl cache;
 
     /**
      * Creates BearerTokenAuthenticationChallengePolicy.
      *
      * @param credential the token credential to authenticate the request
-     * @param scopes the scopes of authentication the credential should get token for
      */
-    public BearerTokenAuthenticationChallengePolicy(TokenCredential credential, String... scopes) {
+    public BearerTokenAuthenticationChallengePolicy(TokenCredential credential) {
         Objects.requireNonNull(credential);
-        this.credential = credential;
-        this.scopes = scopes;
-        this.defaultTokenSupplier = () -> credential.getToken(new TokenRequestContext().addScopes(scopes));
-        this.cache = new AccessTokenCache(defaultTokenSupplier);
+        this.cache = new AccessTokenCacheImpl(credential);
     }
 
     /**
-     *
      * Executed before sending the initial request and authenticates the request.
      *
      * @param context The request context.
      * @return A {@link Mono} containing {@link Void}
      */
     public Mono<Void> onBeforeRequest(HttpPipelineCallContext context) {
-        return authenticateRequest(context, defaultTokenSupplier, false);
+        return Mono.empty();
     }
 
     /**
      * Handles the authentication challenge in the event a 401 response with a WWW-Authenticate authentication
-     * challenge header is received after the initial request.
+     * challenge header is received after the initial request and returns appropriate {@link TokenRequestContext} to
+     * be used for re-authentication.
      *
      * @param context The request context.
      * @param response The Http Response containing the authentication challenge header.
-     * @return A {@link Mono} containing the status, whether the challenge was successfully extracted and handled.
-     *  if true then a follow up request needs to be sent authorized with the challenge based bearer token.
+     * @return A {@link Mono} containing {@link TokenRequestContext}
      */
     public Mono<Boolean> onChallenge(HttpPipelineCallContext context, HttpResponse response) {
```

Comment:
We also renamed this in .NET to `AuthenticateRequestOnChallengeAsync` again to make it clear what implementers are intended to do here.